framework/db/src/com/cloud/utils/db/Transaction.java

```diff
@@ -59,4 +59,30 @@ public T doInTransaction(TransactionStatus status) throws RuntimeException {
         });
     }
 
+    @SuppressWarnings("deprecation")
+    public static <T, E extends Throwable> T execute(final short databaseId, TransactionCallbackWithException<T, E> callback) throws E {
+        String name = "tx-" + counter.incrementAndGet();
+        TransactionLegacy currentTxn = TransactionLegacy.currentTxn(false);
+        short outer_txn_databaseId = (currentTxn != null ? currentTxn.getDatabaseId() : databaseId);
+        TransactionLegacy txn = TransactionLegacy.open(name, databaseId, true);
+        try {
```

Comment:
Just followed the discipline of not disturbing the structure of the code that is outside quota modules.